src/seraphis_wallet/jamtis_keys.h

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2022, The Monero Project
+// 
+// All rights reserved.
+// 
+// Redistribution and use in source and binary forms, with or without modification, are
+// permitted provided that the following conditions are met:
+// 
+// 1. Redistributions of source code must retain the above copyright notice, this list of
+//    conditions and the following disclaimer.
+// 
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list
+//    of conditions and the following disclaimer in the documentation and/or other
+//    materials provided with the distribution.
+// 
+// 3. Neither the name of the copyright holder nor the names of its contributors may be
+//    used to endorse or promote products derived from this software without specific
+//    prior written permission.
+// 
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+// THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+// THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+// NOT FOR PRODUCTION
+
+////
+// Mock jamtis keys
+//
+// reference: https://gist.github.com/tevador/50160d160d24cfc6c52ae02eb3d17024
+///
+
+#pragma once
+
+//local headers
+#include "crypto/blake256.h"
+#include "crypto/chacha.h"
+#include "crypto/crypto.h"
+#include "crypto/x25519.h"
+#include "ringct/rctTypes.h"
+#include "seraphis_core/jamtis_destination.h"
+#include "serialization/keyvalue_serialization.h"
+#include "serialization/serialization.h"
+#include "wipeable_string.h"
+
+//third party headers
+
+//standard headers
+#include <functional>
+#include <vector>
+
+//forward declarations
+
+
+namespace sp
+{
+namespace jamtis
+{
+
+////
+// A set of jamtis keys for mock-ups/unit testing
+///
+struct jamtis_keys
+{
+    crypto::secret_key k_m;           //master
```

Comment:
I don't know the Seraphis lib that closely yet. Are very short and at first sight quite cryptic names like `k_m` the norm there? Do tons of comments refer to this key as `k_m` so it would not be a good idea to give this a more "speaking" name like `master_key`?

Paging @jeffro256, @jberman and @DangerousFreedom1984 for their opinions as well.

Reply:
I agree with this. Imo the key names should be more specific everywhere they're used.

Reply:
it's from jamtis.md  iirc

Reply:
I'm still more a proponent of descriptive names in the code; imo it makes it easier to follow without keeping so much in your head. I think it's fine for the spec (math papers usually use one letter notation). I also think matching the spec is ok, so I'm fine with either way.

Reply:
After thinking about it again, one more vote to switch to the usual descriptive names for things. Devs shouldn't have to deal with names like `xk_fr` or even mixed-case `K_1_base` for so fundamental things. Descriptive names trump having names that are identical with the spec, IMHO.